lib/github-oauth-prompt.js

```diff
@@ -210,6 +209,11 @@ function getOptions (userOptions) {
         throw new Error('Option name must be a non-empty string');
     }
 
+    // Option: host.
+    if (options.host !== undefined && !isNonEmptyString(options.host)) {
```

Comment:
Sure, but we need to check the empty string only if the parameter is present. If we don't, you'll get an error when you don't specify the host. For the other params, you do an undefined check and an assignation, here you only have the check.